systemFlake.nix

```diff
@@ -51,15 +58,32 @@ let
     "checksBuilder"
   ];
 
-  nixosConfigurationBuilder = hostname: profile: (
+  mergeHostAttrs = lhs: rhs: let
+    # convoluted nix-kell, but clean(er) english-german below :-)
+    _ = x: op: y: op x y;
+    oder = lhs: rhs: if lhs != null then lhs else rhs;
+
+    rhs.channelName' = _ rhs.channelName;
+    lhs.channelName' = _ lhs.channelName;
+    rhs.system' = _ rhs.system;
+    lhs.system' = _ lhs.system;
```

Comment:
Now that's interesting :thinking:

Reply:
I've seen such language features in haskell and I was frustrated that I could not do:
`a or b or c`.

Reply:
actually nix has operator overloading! Not sure if it helps here, but you can do things llike `__lessthan = `

Reply:
> I've seen such language features in haskell and I was frustrated that I could not do: `a or b or c`.

Yeah, I realize what you were trying to do here, quite smart tho now that I look at it - at very least I would not mind better naming or comment or two

> actually nix has operator overloading! Not sure if it helps here, but you can do things like __lessthan =
That's cool, I did not know that!

Reply:
Yeah this code is quite smart for what its trying to accomplish

Reply:
I've tried al possible combination. It looks as if overloading `__or` does not as intended.